### PR TITLE
[VSTS][CI] Fix issues with the internal pool and VSTS git usage.

### PR DESCRIPTION
**Pull request by mandel-macaque**
The internal pool, for some reason, do not like the cred store. We need
this to fix the interaction with the maccore repo.

1. VSTS uses https
2. We need to set the pat and try no to modify all bash scripts.

We do so by changing the remote url of maccores git repo so that it
includes the pat.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Sample of it working in the internal pool: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4422839&view=logs&j=600d49c6-e96b-5e8b-2c3a-9f351e2e5848&t=3ef36ff7-7add-52a4-c578-528e39bd10ed

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/automation/templates/build/build.yml [hunk 1]
```diff
@@ -29,10 +29,17 @@ steps:
...
 - pwsh: |
-    rm -Rf "$HOME\.git-credentials"
+    # should we need sudo, no, but someone did something wrong in the images..
+    sudo rm -Rf "$HOME/.git-credentials"
     git config --global credential.helper store
-    Set-Content -Path "$HOME\.git-credentials" -Value "https://$(GitHub.Token):x-oauth-basic@github.com`n" -NoNewline
-  displayName: 'Add git creds store'
+    Set-Content -Path "$HOME/.git-credentials" -Value "https://$(GitHub.Token):x-oauth-basic@github.com`n" -NoNewline
+
+    # maccore is special, we use fetch there in some bash scripts, but VSTS uses https.. and some pools don't likve the above.. :/
```
Comment (spouliot) [inline tools/devops/automation/templates/build/build.yml:37]: typo `lik[v]e`

**Review by spouliot — APPROVED** [review]

File: tools/devops/automation/templates/build/build.yml (hunk 1 above)
Comment (mandel-macaque) [inline tools/devops/automation/templates/build/build.yml:37]:
```suggestion
    # maccore is special, we use fetch there in some bash scripts, but VSTS uses https.. and some pools don't like the above.. :/
```

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**